src/core/src/sim/mc/moves/resize.rs

```diff
@@ -0,0 +1,112 @@
+// Cymbalum, an extensible molecular simulation engine
+// Copyright (C) 2015-2016 G. Fraux — BSD license
+
+use rand::distributions::{Sample, Range};
+use rand::Rng;
+
+use std::f64;
+use std::mem;
+
+use super::MCMove;
+
+use types::{Matrix3, One};
+use sys::{System, EnergyCache};
+
+/// Monte-Carlo move that changes the size of the simulation cell
+pub struct Resize {
+    /// Delta for translation of the box length
+    delta: f64,
+    /// Sampling range for volume scaling
+    range: Range<f64>,
+    /// System before applying changes to the simulation box
+    old_system: System,
+    /// target pressure
+    pressure: f64,
+    /// largest cut off diameter of `PairPotentials`
+    rc_max: f64,
+}
+
+impl Resize {
+    /// Create a new `Resize` move, with target pressure `pressure` and maximum
+    /// displacement of `delta`.
+    pub fn new(pressure: f64, delta: f64) -> Resize {
+        assert!(delta > 0.0, "delta must be positive in Resize move");
+        // TODO: set rc_max from the largest cut off in PairPotentials
+        Resize {
+            delta: 0.,
+            range: Range::new(-delta, delta),
+            old_system: System::new(),
+            pressure: pressure,
+            rc_max: 100.0, // arbitrarily large value
```

Comment:
I recognized that there has to be a zero. `std::f64::INFINITY` would lead to the error, since any side length is smaller than infinity.